journal/src/main/java/io/zeebe/journal/file/record/PersistedJournalRecord.java

```diff
@@ -15,63 +15,79 @@
  */
 package io.zeebe.journal.file.record;
 
-import com.google.common.base.Objects;
 import io.zeebe.journal.JournalRecord;
+import java.nio.ByteBuffer;
+import java.util.Objects;
 import org.agrona.DirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
 
-/** Journal Record */
-public final class PersistedJournalRecord implements JournalRecord {
+/**
+ * A JournalRecord stored in a buffer.
+ *
+ * <p>A {@link PersistedJournalRecord} consists of two parts. The first part is {@link
+ * PersistedJournalRecordMetadata}. The second part is {@link PersistedJournalIndexedRecord}.
+ */
+public class PersistedJournalRecord implements JournalRecord {
+  final PersistedJournalRecordMetadata metadata;
+  final PersistedJournalIndexedRecord record;
 
-  private final DirectBuffer data;
-  private final long index;
-  private final long asqn;
-  private final int checksum;
+  public PersistedJournalRecord(final ByteBuffer buffer) {
+    final var slice = buffer.slice();
```

Comment:
I would pass in a direct buffer directly, or at the very least a slice and/or read only buffer (probably more a `asReadOnlyBuffer` since we don't expect to modify it anyway). But probably a `DirectBuffer` directly, as I'd like to use as little of `ByteBuffer` as possible to avoid the error prone API.

If we want to be "optimized", probably want to pass in the original buffer, then the record's offset in it. We can then pass that to the metadata, then pass `offset + metadata.getLength()` and the original buffer to the data itself. I'm fine with creating additional buffer views if it makes things simpler.